core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql

```diff
@@ -44,16 +44,22 @@
              to_relation=target_relation) %}
     {#-- Process schema changes. Returns dict of changes if successful. Use source columns for upserting/merging --#}
     {% set dest_columns = process_schema_changes(on_schema_change, temp_relation, existing_relation) %}
-    {% if not dest_columns %}
-      {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
-    {% endif %}
 
-    {#-- Get the incremental_strategy, the macro to use for the strategy, and build the sql --#}
-    {% set incremental_strategy = config.get('incremental_strategy') or 'default' %}
-    {% set incremental_predicates = config.get('incremental_predicates', none) %}
-    {% set strategy_sql_macro_func = adapter.get_incremental_strategy_macro(context, incremental_strategy) %}
-    {% set strategy_arg_dict = ({'target_relation': target_relation, 'temp_relation': temp_relation, 'unique_key': unique_key, 'dest_columns': dest_columns, 'predicates': incremental_predicates }) %}
-    {% set build_sql = strategy_sql_macro_func(strategy_arg_dict) %}
+    {% if dest_columns is mapping and dest_columns.get("full_refresh") %}
+      {% set build_sql = get_create_table_as_sql(False, intermediate_relation, sql) %}
+      {% set need_swap = true %}
+    {% else %}
```

Comment:
We'll need logic like this in every `incremental` materialization, including the modified versions in other adapter plugins we maintain:
- [dbt-snowflake](https://github.com/dbt-labs/dbt-snowflake/blob/e18c501175c84d5b08d66244533e636a5a04dba1/dbt/include/snowflake/macros/materializations/incremental.sql#L74-L75)
- [dbt-bigquery](https://github.com/dbt-labs/dbt-bigquery/blob/da0d0d7e21c0bec09a9fd35baa264f915eddc55a/dbt/include/bigquery/macros/materializations/incremental.sql#L130-L131)
- [dbt-spark](https://github.com/dbt-labs/dbt-spark/blob/5ca20be56ec2d557b4fff5e42c320949040650d3/dbt/include/spark/macros/materializations/incremental/incremental.sql#L56)